Talk chat bot for taxi

hihilisk opened the pull request.

hihilisk committed:
Corrected code: added returning a boolean variable.

hihilisk committed:
fixe tabulation

hihilisk committed:
code alignment to original requirements

hihilisk committed:
refactoring code

hihilisk committed:
remove 'return'

hihilisk committed:
new task:
check string on the palindrome
check the string without spaces on the palindrome

hihilisk committed:
refactoring code

hihilisk committed:
(added folder`s to ingore entries)

hihilisk committed:
refactoring code v2

hihilisk committed:
created function palindrome on JS

hihilisk committed:
refactoring code

hihilisk committed:
refactoring code

hihilisk committed:
refactoring code palindrome.rb

hihilisk committed:
refactoring code palindrome.js
renamed the variable 'string' to 'value'
returned spaces to operands
removed unnecessary 'else'

hihilisk committed:
create sorting function

hihilisk committed:
refactori sort_bubbles.js he`s work

hihilisk committed:
Revert "refactori sort_bubbles.js he`s work"

This reverts commit 449fb604

hihilisk committed:
refactori sort_bubbles.js he`s work

hihilisk committed:
refactori sort_bubbles.js he`s exactly work

hihilisk committed:
refactori sort_bubbles.js sort_bubbles.rb

hihilisk committed:
rename argument 'unsorted_array' 'array'

hihilisk committed:
first commit

hihilisk committed:
delete trash

hihilisk committed:
backup progress, at create talking_bot

hihilisk committed:
backup progress, at create talking_bot

hihilisk committed:
backup progress, at refactoring bot_for_a_taxi

hihilisk committed:
work version

hihilisk committed:
create validation date order

hihilisk committed:
worked version

hihilisk committed:
fix behavior method`s 'validation_entered_date'

alexander-maxa commented on bot_for_a_taxi/bot_for_a_taxi.rb:
```diff
@@ -0,0 +1,155 @@
+  def initialize(from = 0, into = 0)
+    @from = 0
+    @into = 0
+  end
```
У тебя конструктор принимает два параметра, которые ты потом не используешь. Складывается впечатление, что ты не понимаешь, как и для чего нужны параметры в методе. А главное - для чего нужен конструктор (

alexander-maxa replied:
Почему в конструкторе происходит вызов методов?

hihilisk replied:
почему нельзя вызывать методы в конструкторе?

hihilisk committed:
refactoring code: all methods, delete 'chat_bot', delete unnecessary methods

hihilisk committed:
refactoring code:
  -delete method "reset"
  -added "attr_reader" giving access to variables
  -all methods with the exception of 'initialize' moved in 'private' scope
  Optimize:
    -methods no more call methods, they work only with variables

refactoring method "initialize":
  -now he called all methods in the right sequence

alexander-maxa commented on bot_for_a_taxi/bot_for_a_taxi.rb:
```diff
@@ -0,0 +1,49 @@
+  end
+
+  def request_average_speed
+    @average_speed = 60.0
```
random

alexander-maxa commented on bot_for_a_taxi/bot_for_a_taxi.rb:
```diff
@@ -0,0 +1,49 @@
+  end
+
+  def calculate_coast
+    rate = 8
```
В константу

alexander-maxa commented on bot_for_a_taxi/bot_for_a_taxi.rb:
```diff
@@ -0,0 +1,49 @@
+  end
+
+  def choice_taxi_car
+    taxi_car_location = { 1 => { model: 'Hyundai solaris', color: 'серая', number: 'x161xx161', distance: 5 },
```
Переделать на обычный массив, еще мы индексы руками не писали :)

hihilisk replied:
подразумевалось что существует некая база которая занимается отслеживанием местоположения а/м и для удобства каталогизации хранимых данных присваивать им id ), я переделал но суть замечания не понял до конца

hihilisk committed:
Delete totalizer.js

hihilisk committed:
Delete totalizer.rb

hihilisk committed:
refactoring code:
  -delete method "request_average_speed"
  -crate const 'AVERAGE_SPEED'
    choice_taxi_car
  -rename unused variable '_key' to 'key'
    calculate_coast
  -randomized value variable 'rate'
refactoring methods: calculate_time_waiting and calculate_time_trip
 -changes made for use new const 'AVERAGE_SPEED'

hihilisk committed:
Merge branch 'bot_for_a_taxi' of https://github.com/hihilisk/study into bot_for_a_taxi

hihilisk committed:
refactoring code:
    choice_taxi_car
  -change attachment 'hesh in hesh' to 'hesh in array'
  -deleted no longer used actions with variable '@taxi_car'

alexander-maxa commented on bot_for_a_taxi/bot_for_a_taxi.rb:
```diff
@@ -0,0 +1,44 @@
+  def initialize(from, into)
+    @from = from
+    @into = into
+    calculate_distance
```
Убрать отсюда от 8 строчки, до 12. Эти методы должны вызываться только тогда, когда нужно. Когда на экземпляре класса ты их будешь вызывать

alexander-maxa replied:
Конструктор просто создает объект. Конструткор - констроировать, билдить.

hihilisk committed:
refactoring code:
    initialize
  -deleted all calling methods
  -add calculate "@distance_trip"
    new methods
  -price: calling private method 'calculate_coast' and pasting result in string
  -distance: calling variable '@distance_prim' and pasting result in string
  -waiting_car_time: calling private method 'select_car' with key [:distance], const 'AVERAGE_SPEED' and pasting result in string
    calculate_coast and select_car
  -moving in private scope

hihilisk committed:
refactoring code:
    initialize
  -result calculate "@distance_trip" pasted in string
    price
  -calculated cost and pasting result to string
  -removed method 'calculate_coast'
  -added access to variable "@distance" with method 'attr_reader'

hihilisk committed:
refactoring code

alexander-maxa commented on bot_for_a_taxi/bot_for_a_taxi.rb:
```diff
@@ -0,0 +1,30 @@
+  def initialize(point_a, point_b)
+    @point_a = point_a
+    @point_b = point_b
+    @distance = "#{(@point_a - @point_b).abs} km"
```
def distance
 "#{(@point_a - @point_b).abs}"
end

alexander-maxa commented on bot_for_a_taxi/bot_for_a_taxi.rb:
```diff
@@ -0,0 +1,30 @@
+  end
+
+  def price
+    @coast = "#{((select_car[:distance] + @distance.to_i) / RATE).round(2)} $"
```
@distance.to_i -> distance

alexander-maxa replied:
to_i не нужно делать, должны цифры приходить уже

hihilisk replied:
@distance строка, посему она явно преобразовывалась в число

hihilisk committed:
Refactoring code.
  class TaxiBot:
    -created method 'operator':
      he calling 'Operator.new'

Created class 'Operator'

hihilisk committed:
Refactoring: method 'operator' return array, containing data about the operator

hihilisk committed:
Refactoring_code:
  Creating class 'Operator':
    Creating methods:
      -initialize
      -methods to access reading variables
  File 'bot_for_a_taxi.rb':
    -requested class 'Operator'
    -creating methods 'creating_operator'
    -create instance variable '@operator_taxi'

alexander-maxa commented on bot_for_a_taxi/bot_for_a_taxi.rb:
```diff
@@ -0,0 +1,63 @@
+    @operator_taxi
+  end
+
+  def creating_operator(name = nil, age = nil, sex = nil, avatar_img = nil)
```
create_operator

alexander-maxa commented on bot_for_a_taxi/bot_for_a_taxi.rb:
```diff
@@ -0,0 +1,63 @@
+  end
+
+  def taxi_car
+    selecting_taxi
```
select_taxi

hihilisk committed:
Refactoring code:
  bot_for_a_taxi.rb

    renaming:
      -'selecting_taxi' to 'select_taxi'
      -'taxi_location' to 'taxi_base'

    method:
      -created 'create_operator'
      -'operator' return operator if arguments were not passed, call method 'create_operator'

alexander-maxa commented on bot_for_a_taxi/class_operator.rb:
```diff
@@ -0,0 +1,26 @@
+    @sex
+  end
+
+  def avatar_img
```
avatar_url
